src/tests/ERC20.t.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.0;
+
+import "../TokenFuzzChecks.sol";
+import "./SampleToken.sol";
+
+contract ERC20Test is TokenChecks, TokenFuzzChecks {
+
+    address token;
+    string contractName = "SampleToken";
+    string tokenName = "Sample Token";
+    string symbol = "TKN";
+    string version = "1";
+    uint8 decimals = 18;
+
+    function setUp() public {
+        token = address(new SampleToken());
+        TokenLike(token).deny(address(this));
+    }
+
+    // Unit Tests
+
+    function test_checkERC20() public {
+        checkERC20(token, contractName, tokenName, symbol, version, decimals);
```

Comment:
This in practive will make the erc20 checks run twice right?
Do you think we can just remove checkERC20? or otherwise if you still think it's useful just don't have a test that calls it?

Reply:
Yes, good point, I'll remove those calls from TokenTests.sol so they can still be called explicitly for setup (2) but won't be called by default in setup (1).

Reply:
I've added tests for the bulk checks to TokenTest.t.sol in f0e734f88d80d051bf7414523d4070c0e850c155. Without these, there would be no check that the check functions have been correctly made path independent for these bulk checks to run without issue.